src/classes/RD2_ScheduleService.cls

```diff
@@ -0,0 +1,120 @@
+
+    private void init(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
+        parentRecurringDonation = rd;
+        if (rdSchedules != null) {
+            recurringDonationSchedules = rdSchedules;
+        }
+    }
+
+    public RD2_ScheduleService(npe03__Recurring_Donation__c rd) {
+        init(rd, null);
+    }
+
+    public RD2_ScheduleService(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
+        init(rd, rdSchedules);
+    }
+
+    /*******************************************************************************************************
+    * @description Initializes RecurringDonationSchedule__c record(s) associated with a parent
+    * npe03__Recurring_Donation__c record (parentRecurringDonation)
+    * @return List<RecurringDonationSchedule__c>
+    */
+    public List<RecurringDonationSchedule__c> initializeSchedule() {
+        if (parentRecurringDonation != null && parentRecurringDonation.Id != null && parentRecurringDonation.Status__c != 'Closed') {
+            recurringDonationSchedules.add(new RecurringDonationSchedule__c());
+            if (parentRecurringDonation.Day_of_Month__c == '1st and 15th' && parentRecurringDonation.npe03__Installment_Period__c == 'Monthly') {
+                recurringDonationSchedules.add(new RecurringDonationSchedule__c());
+            }
+        }
+        generateRecurringDonationScheduleDetails();
+        return recurringDonationSchedules;
+    }
+
+    // TODO: Update schedule to reflect parentRecurringDonation - W-036113
+    public List<RecurringDonationSchedule__c> updateSchedule() {
+        return null;
+    }
+
+    // TODO: Will return next Donation date - W-036251
+    public Date getNextDonationDate() {
+        return null;
+    }
+
+    // TODO: Will return an inner class (TBD)
+    public List<Date> visualizeSchedule() {
+        return null;
+    }
+
+    private void generateRecurringDonationScheduleDetails() {
+        Integer recordCount = 1;
+        for (RecurringDonationSchedule__c rds : recurringDonationSchedules) {
+            rds.InstallmentAmount__c = parentRecurringDonation.npe03__Installment_Amount__c;
+            rds.InstallmentFrequency__c = parentRecurringDonation.InstallmentFrequency__c;
+            rds.InstallmentPeriod__c = parentRecurringDonation.npe03__Installment_Period__c;
+            rds.PaymentMethod__c = parentRecurringDonation.PaymentMethod__c;
+            rds.RecurringDonation__c = parentRecurringDonation.Id;
+            rds.Campaign__c = parentRecurringDonation.npe03__Recurring_Donation_Campaign__c;
+            Date refDate =
+                parentRecurringDonation.StartDate__c > parentRecurringDonation.npe03__Date_Established__c ?
+                parentRecurringDonation.StartDate__c :
+                parentRecurringDonation.npe03__Date_Established__c;
+            if (parentRecurringDonation.npe03__Installment_Period__c == 'Monthly') {
```

Comment:
Same with the date logic here. Even though everything in the `if` is abstracted out, the logic here would also be separated out into a smaller method that handles the different date calculations.